packages/core/src/common/runnable/helpers.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import type { DiContainerForInjection, Injectable, InjectionInstanceWithMeta, InjectionToken } from "@ogre-tools/injectable";
+import { getOrInsertSetFor, isDefined } from "../utils";
+import * as uuid from "uuid";
+import assert from "assert";
+import type { Runnable, RunnableSync, RunnableSyncWithId, RunnableWithId } from "./types";
+
+const computedNextEdge = (traversed: string[], graph: Map<string, Set<string>>, currentId: string, seenIds: Set<string>) => {
+  seenIds.add(currentId);
+  const currentNode = graph.get(currentId);
+
+  assert(currentNode, `Runnable graph does not contain node with id="${currentId}"`);
+
+  for (const nextId of currentNode.values()) {
+    if (traversed.includes(nextId)) {
+      throw new Error(`Cycle in runnable graph: "${traversed.join(`" -> "`)}" -> "${nextId}"`);
+    }
+
+    computedNextEdge([...traversed, nextId], graph, nextId, seenIds);
+  }
+};
+
+export function verifyRunnablesAreDAG<Param>(injectionToken: InjectionToken<Runnable<Param>, void>, runnables: RunnableWithId<Param>[]): void;
```

Comment:
It should be documented that DAG refers to (I assume) directed acyclic graph.

Reply:
That is correct and sure will do.